README.md

```diff
@@ -23,68 +23,77 @@ Never submit security-related bugs through a Github Issue or by email.
+|----------------------|---------------------------------------------------------------------------------|
+| yarn dev:admin       |  Starts the dev server (admin app)                                              |
+| yarn dev:amo         |  Starts the dev server and proxy (amo)                                          |
+| yarn dev:amo:no-proxy|  Starts the dev server without proxy (amo)                                      |
+| yarn dev:disco       |  Starts the dev server (discovery pane)                                         |
+| yarn flow:check      |  Check for Flow errors and exit                                                 |
+| yarn flow:dev        |  Continuously check for Flow errors                                             |
+| yarn eslint          |  Lints the JS                                                                   |
+| yarn stylelint       |  Lints the SCSS                                                                 |
+| yarn lint            |  Runs all the JS + SCSS linters                                                 |
+| yarn version-check   |  Checks you have the required dependencies                                      |
+| yarn test            |  Enters [jest][] in watch mode                                                  |
+| yarn test-ci         |  Runs all continuous integration checks. This is only meant to run on TravisCI. |
+| yarn test-coverage   |  Runs all tests and reports code coverage                                       |
+| yarn test-once       |  Runs all tests with [jest][] and exits                                         |
 
 ### Running tests
 
-You can run the entire test suite with `npm test` but there are a few other ways
-to run tests.
+You can enter the interactive [jest][] mode by typing `yarn test`.
+This is the easiest way to develop new features.
 
-#### Run all unit tests in a loop
+Here are a few tips:
 
-You can use `npm run unittest:dev` to run all unit tests in a loop while you
-edit the source code.
+* When you start `yarn test`, you can switch to your code editor and begin
+  adding test files or changing existing code. As you save each file, [jest][]
+  will only run tests related to the code you change.
+* If you had typed `a` when you first started then [jest][] will continue to
+  run the full suite even when you change specific files. Type `o` to switch
+  back to the mode of only running tests related to the files you are changing.
+* If you see something like `Error watching file for changes: EMFILE` on Mac OS
+  then `brew install watchman` might fix it.
+  See https://github.com/facebook/jest/issues/1767
 
-#### Run a subset of the unit tests
+#### Run a subset of the tests
 
-If you don't want to run the entire unit test suite, first you have to start a
-unittest server:
+By default, `yarn test` will only run a subset of tests that relate to the code
```

Comment:
As mentioned, that depends. It seems to default to running anything it finds since the last commit. But if there is nothing it just sits there waiting for a prompt.